components/account-details.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <div class="account-details">
+    <h3 class="h4">
+      Account Details
+    </h3>
+    <dl v-if="user" class="account-details__list">
+      <dt>Name</dt>
+      <dd>{{ name }}</dd>
+      <dt>Email address</dt>
+      <dd>{{ email }}</dd>
+      <dt>Phone number</dt>
+      <dd>{{ phone }}</dd>
+    </dl>
+    <div v-else class="account-details__placeholder">
+      <p>Please login to view your account details</p>
+    </div>
+    <div class="account-details__buttons">
+      <UiButton v-if="!user" kind="primary" @click="login">
+        Log in
+      </UiButton>
+      <UiButton v-else kind="primary" @click="logout">
+        Log out
+      </UiButton>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapState } from 'vuex'
+import auth from '../auth'
+import UiButton from './ui-button'
+
+export default {
+  components: { UiButton },
+  computed: {
+    ...mapState('preferences', ['user']),
+    name () {
+      return this.user.name || '--'
```

Comment:
Shouldn't this ( || ) or-logic be in the vuex store? also for email and phone

Reply:
No the store should just saves the `user` object, what you do with its content / how you use it is up the component. If `user` does not exist (`null`) the component shows a `--` as a placeholder. That's not up to the store to decide. If use `user` in a different component you might not want `--` to be returned.